packages/blade/src/components/Spinner/BaseSpinner/SpinningBox.native.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import type { EasingFunctionFactory } from 'react-native-reanimated';
+import Animated, {
+  withTiming,
+  cancelAnimation,
+  useAnimatedStyle,
+  useSharedValue,
+  withRepeat,
+} from 'react-native-reanimated';
+import { motion as spinnerMotion } from './spinnerTokens';
+import { useTheme } from '~components/BladeProvider';
+import { getIn, makeMotionTime } from '~utils';
+import Box from '~components/Box';
+
+const SpinningBox = ({ children }: { children: React.ReactNode }): React.ReactElement => {
+  const { theme } = useTheme();
+  const duration = (makeMotionTime(
+    getIn(theme.motion, spinnerMotion.duration),
+  ) as unknown) as number;
+  const easing = getIn(theme.motion, spinnerMotion.easing) as EasingFunctionFactory;
+
+  const rotation = useSharedValue(0);
+  const animatedStyles = useAnimatedStyle(() => {
+    return {
+      transform: [
+        {
+          rotateZ: `${rotation.value}deg`,
+        },
+      ],
+    };
+  }, [rotation.value]);
+
+  React.useEffect(() => {
+    rotation.value = withRepeat(
+      withTiming(360, {
+        duration,
+        easing,
+      }),
+      -1,
+    );
+    return (): void => {
+      cancelAnimation(rotation);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can cause potential stale closure issues

Reply:
Hmmm. What else could we do here? We want it to be triggered on mount and cancel the animation on unmount

Reply:
Either using refs inside or passing the variables in dependency array, whichever would make more sense and looks more readable.

Reply:
Added